physics/GFS_GWD_generic.F90

```diff
@@ -117,15 +121,20 @@ subroutine GFS_GWD_generic_pre_run(                               &
         elvmax = 0
       endif   ! end if_nmtvr
 
-      if (lssav) then
-        if (ldiag3d .and. flag_for_gwd_generic_tend) then
-          do k=1,levs
-            do i=1,im
-              dt3dt(i,k) = dt3dt(i,k) - dtdt(i,k)*dtf
-              du3dt(i,k) = du3dt(i,k) - dudt(i,k)*dtf
-              dv3dt(i,k) = dv3dt(i,k) - dvdt(i,k)*dtf
-            enddo
-          enddo
+      if (lssav .and. ldiag3d .and. flag_for_gwd_generic_tend) then
```

Comment:
Here we don't care  about "all" tendencies, just the GWD ones. We should be passing in the relevant slices of the tendency arrays one at a time using pre-computed indices.

Reply:
But looking at all the other files below that fall into the same category, this gets very complicated and lengthy. Given that this tendency code will be replaced with a capgen solution in the future, ok by me for the moment.

Reply:
Changing that would complicate the code a lot. This is intended to be a medium-term solution, until the CCPP framework is replaced. If the current framework is kept long-term, we can evolve this solution towards a model-independent one. That won't be simple though.